### PR TITLE
Draft 3.0 release notes

### PR DESCRIPTION
**Pull request by MridulS**

### CODE REVIEW

**Comment by jarrodmillman** [comment]
@dschult @rossbar @MridulS I normally make the final updates to the release notes when I make the release. Since this PR already renames and the release notes, let's not merge this until I make the release. For now, if anyone else needs to update the release notes, they should make a PR to Mridul's branch. I will make the final pushes to this PR when I make the release and merge it.

No one should edit `doc/release/release_dev.rst` now, since it will cause conflicts with this PR.

File: doc/release/release_3.0.rst [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+- Only run scheduled pytest-randomly job in main repo. (#5993)
+- Fix steiner tree test (#5999)
+- Add Tidelift security vulnerability link (#6001)
+- Update linters (#6006)
+- Update doc requirements (#6008)
+- VF2++ for Directed Graphs (#5972)
+- Fix defect and update docs for MappedQueue, related to gh-5681 (#5939)
+- Fix warnings from running tests in randomized order (#6014)
+- Update pydata-sphinx-theme (#6012)
+- update secutiry link to tidelift (#6019)
```
Comment (dschult) [inline doc/release/release_3.0.rst:149]:
```suggestion
- update security link to tidelift (#6019)
```

File: doc/release/release_3.0.rst [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+- Update linters (#6024)
+- Fixed test for average shortest path in the case of directed graphs (#6003)
+- Minor updates to expanders generator tests (#6027)
+- Update deprecations after 3.0 dep sprint (#6031)
+- Use scipy.sparse array datastructure (#6037)
+- Designate 3.0b1 release
+- Bump release version
+- Use org funding.yml
+- Add missing asserts to tests (#6039)
+- fixes #6036 (#6080)
```
Comment (dschult) [inline doc/release/release_3.0.rst:161]:
```suggestion
- improve test coverage for algorithms in load centrality (#6080)
```

File: doc/release/release_3.0.rst [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+- Test on Python 3.11 (#6159)
+- Improve test coverage in algorithms shortest paths unweighted.py (#6121)
+- Increased test coverage algorithms/matching.py (#6095)
+- Renamed test functions in test_lowest_common_ancestors (#6110)
+- Increase covering coverage (#6099)
+- Add example for fiedler_vector (#6155)
+- Improve test coverage for cycles.py (#6152)
+- Added an example in all_pairs_node_connectivity  (#6126)
+- Amount of nodes and edges have mistakes when reading adjlist file (#6132)
+- Update pytest (#6165)
```
Comment (dschult) [inline doc/release/release_3.0.rst:192]:
```suggestion
- Update pytest to v7.2 (#6165)
```

File: doc/release/release_3.0.rst [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+- Improve test coverage for voterank algorithm (#6161)
+- plugin based backend infrastructure to use multiple computation backends (#6000)
+- Undocumented parameters in dispersion (#6183)
+- Swap.py coverage to 100 (#6176)
```
Comment (dschult) [inline doc/release/release_3.0.rst:196]:
```suggestion
- improve swap.py test coverage  (#6176)
```

File: doc/release/release_3.0.rst [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+- Improve test coverage expanders stochastic graph generators (#6073)
+- Update developer requirements  (#6194)
+- Designate 3.0rc1 release
+- Bump release version
+- Tests added in test_centrality.py (#6200)
+- add laplacian_spectrum example (#6169)
+- PR for issue #6033 Improve test coverage for algorithms in betweenness_subset.py #6033 (#6083)
+- Di graph edges doc fix (#6108)
+- Improve coverage for core.py (#6116)
+- Add clear edges method to the list of methods to be frozen by the nx.… (#6190)
```
Comment (dschult) [inline doc/release/release_3.0.rst:212]:
```suggestion
- Add clear_edges() method to the list of methods to be frozen by nx.freeze() (#6190)
```

**Review by dschult — COMMENTED** [review]
I'm sure we will have to add some to the end of this list. But it looks good to me tat the moment. I've got a few typos and small changes to the commit messages below.

**Comment by dschult** [comment]
It looks like the `contribs.py` script which reports all the commits, etc is not collecting the number of commits correctly. Or maybe a better way to say it is that since v2.8.5 all the releases report having the same number of commits. The offending command is:  `git rev-list networkx-2.8.8..HEAD --count`.  It works fine up until 2.8.5 and then gets stuck reporting 219 for all the tags 2.8.5-8.  I think Jarrod mentioned that something was no longer being updated with each release -- maybe a tag was not created?  I can't remember the specifics. 

Perhaps we should remove that line from the release notes -- or figure out a way to count the releases between these releases.

**Comment by MridulS** [comment]
I fixed this locally but didn't push the updated the contribs.py as I haven't really tested it properly. The release notes for 2.8.5 should be fine in this PR.
The 3.0 ones are based on top of 2.8.5 currently, should I do it on top of 2.8.8 or 2.8?

**Comment by MridulS** [comment]
I've pushed the updated contribs.py file too, it isn't really tested much but it can be used for generating notes by comparing 2 old releases.

**Review by jarrodmillman — APPROVED** [review]

**Comment by jarrodmillman** [comment]
I removed the 2.8.6, 2.8.7, and 2.8.8 release notes since they weren't consistent with the actuall 2.8.6, 2.8.7, and 2.8.8 release notes. I am not sure what the issue was but the ones here included things that weren't in the releases. I just need to grab them from the 2.8 release branch, which was the original plan for this release. I also converted the draft release notes back to a dev name, since I normally make the name change at the last moment as part of the release process. I will need to update the release notes then anyway.

For the next release, we should stick with the old pattern of just updating the relevant sections in the draft and wait to finalize the release notes and run the contrib.py script when it is time. There is no advantage to run it more than once.

I kept the changes to contrib.py for now, but will review whether the scripts still works as expected when I make the release.

**Comment by jarrodmillman** [comment]
Once the tests pass, this is ready to merge and I can start the release process. Thanks!

**Pull request merged**